### ISSUE TITLE
Change legend colour blocks from spans to buttons and use aria-describedby

### ISSUE THREAD

**Issue by LucyGwilliamAdmin** [open-sdg/open-sdg#867]
![image](https://user-images.githubusercontent.com/43404358/91987617-357ac580-ed26-11ea-9410-42079861a443.png)

Spoke to DAC about legend text placement this morning, and described issue with text being in downloaded image if text moved right by legend so they suggested this alternative:

Legend colour blocks are currently spans instead of buttons:
https://github.com/open-sdg/open-sdg/blob/master/_includes/assets/js/indicatorView.js#L388.

These should be buttons instead, and should use `aria-describedby="legend-help"` for the buttons.

They said this alternative could still be an issue for low-vision users as if they're really zoomed in they won't see the text, so would still need to move closer to legend, @AnnCorp suggested perhaps under footnote so not in capture area for downloaded image? @phillipgardiner what are your thoughts from a design p.o.v?

**Comment by phillipgardiner** [comment]
@brockfanning @LucyGwilliamAdmin @AnnCorp If they were to be buttons, would they function like a buttons? as this would be a bit odd, as they'd be clickable but not do anything.  Is there not a way programmatically of saying if the download image button is pressed then to ignore the line of code that says about the legend? thus is does not display in the image That way it does not matter where it is placed?

**Comment by brockfanning** [comment]
I believe it's technically possible to hide certain elements in the downloaded image, yes. If I understand right then I should give this a shot first, and if that fails then use aria-describedby.

I don't totally understand the span vs. button change, but I will go ahead and give it a try. My interpretation is that buttons have a semantic meaning (something to activate to do something) whereas spans do not.

**Comment by brockfanning** [comment]
I've submitted a PR at #878. Here are the changes:
1. Legend help text moved to just above the legend
2. Legend items wrapped in `<button>` elements
3. Legend has `aria-labelledby` referring to the legend help
4. Legend items have `aria-pressed` which changes as user clicks
5. Legend items have `aria-label` with "Press to hide the dataset x" (where "x" is the name of the dataset, eg "Rural", etc.)
6. Legend help is hidden from image download
